js/customize-snapshots-front.js

```diff
@@ -0,0 +1,72 @@
+/* global jQuery, wp */
+/* exported CustomizeSnapshotsFront */
+var CustomizeSnapshotsFront = (function( $ ) {
+	'use strict';
+
+	var component = {
+		data: {
+			confirmationMsg: '',
+			action: '',
+			snapshotsFrontendPublishNonce: '',
+			errorMsg: ''
+		}
+	};
+	/**
+	 * Init.
+	 *
+	 * @param {object} args Args.
+	 * @returns {void}
+	 */
+	component.init = function init( args ) {
+		_.extend( component.data, args );
+
+		$( document ).ready( function() {
+			component.setupPublishButton();
+		} );
+	};
+
+	/**
+	 * Set up snapshot frontend publish button.
+	 *
+	 * @returns {void}
+	 */
+	component.setupPublishButton = function setupPublishButton() {
+		var publishBtn = $( '#wp-admin-bar-publish-customize-snapshot a' );
+
+		if ( ! publishBtn.length ) {
+			return;
+		}
+
+		publishBtn.click( function( e ) {
+			var request,
+				data = {
+					nonce: component.data.snapshotsFrontendPublishNonce,
+					uuid: component.data.uuid
+				};
+			e.preventDefault();
+
+			if ( ! window.confirm( component.data.confirmationMsg ) ) { // eslint-disable-line no-alert
+				return false;
+			}
+			if ( ! wp.customize.settings.theme.active ) {
```

Comment:
@westonruter After merging the develop there is an issue with publishing changeset from preview:
![screen shot 2017-07-20 at 5 53 33 pm](https://user-images.githubusercontent.com/3294597/28423742-8f73d480-6d74-11e7-8d58-21924a43150c.png)

Looks like `wp.customize` is not defined at this moment. Do you know if anything changed meanwhile in the logic or order of enqueuing the JS scripts? Tried to make the script dependent on `customize-base` but in this case `wp.customize.settings` is undefined.

Would be great if you could point out what to investigate or what might have changed to cause this issue. Not sure what I'm missing here, maybe it's something very obvious.